pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -1368,16 +1377,8 @@ func (gce *GCECloud) NodeAddresses(_ string) ([]api.NodeAddress, error) {
 	}, nil
 }
 
-func (gce *GCECloud) isCurrentInstance(instance string) bool {
-	return gce.instanceID == canonicalizeInstanceName(instance)
-}
-
 // ExternalID returns the cloud provider ID of the specified instance (deprecated).
 func (gce *GCECloud) ExternalID(instance string) (string, error) {
-	// if we are asking about the current instance, just go to metadata
```

Comment:
Added back in logic to try metadata server first, and fall back to API call only if it fails (or GCECloud object is specifically setup indicating no metadata server support).